compiler/driver.c

```diff
@@ -567,6 +579,7 @@ extern int main(int argc, char * argv[]) {
             lose_b(p->b); FREE(p); p = q;
         }
     }
+    free((void*)o->name);
```

Comment:
Sorry, this is just too nasty.

I think you're doing this to cast away `const`-ness so the pointer can be passed to `free()`, but really if we want to be able to call `free()` we shouldn't make the pointer `const` to start with.  That way if somebody tries to do `o->name = "some_constant_string";` it'll fail to compile, but as you have this it'll compile and only fail at run-time, and only if that code path is actually taken.